pulsar-broker/src/main/java/org/apache/pulsar/compaction/TwoPhaseCompactor.java

```diff
@@ -91,6 +91,9 @@ protected CompletableFuture<Long> doCompaction(RawReader reader, BookKeeper bk)
     private CompletableFuture<PhaseOneResult> phaseOne(RawReader reader) {
         Map<String, MessageId> latestForKey = new HashMap<>();
         CompletableFuture<PhaseOneResult> loopPromise = new CompletableFuture<>();
+        FutureUtil.addTimeoutHandling(loopPromise,
```

Comment:
maybe `30s` is not enough for the step one

Reply:
It does, it is predictable for the time of reading one single message, but not for executing whole phase one.